src/TicketingSystem.DataAccess/Entities/BaseEntity.cs

```diff
@@ -0,0 +1,13 @@
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using System;
+
+namespace TicketingSystem.DataAccess.Entities
+{
+    public abstract class BaseEntity : IStringKeyEntity
+    {
+        [BsonId]
```

Comment:
BaseEntity -> IStringKeyEntity -> IEntity<string> Do we really need this inheritance chain just to resolve Id? (YAGNI) Most likely we will not have anything else here. I think something like this would be enough:
```
public interface IHasId
{
    public string Id { get; set; }
}
```

Reply:
Fixed